src/display/text_layer.js

```diff
@@ -528,20 +534,48 @@ var renderTextLayer = (function renderTextLayerClosure() {
     },
 
     _render: function TextLayer_render(timeout) {
-      var textItems = this._textContent.items;
-      var textStyles = this._textContent.styles;
-      for (var i = 0, len = textItems.length; i < len; i++) {
-        appendText(this, textItems[i], textStyles);
+      let capability = createPromiseCapability();
+      let pump;
+      if (this._textContent === undefined) {
+        pump = () => {
+          reader.read().then(({ value, done, }) => {
+            if (done) {
+              capability.resolve();
+              return;
+            }
+            Util.appendToArray(this._textContent.items, value.items);
```

Comment:
We don't need to collect all items from all chunks. We can process items as they coming in. Create e.g. `processItems(items, stylesCache)` and use it for a chunk and "this._textContent" case: move appendText-loop logic there. For chunks you have to maintain stylesCache with all styles, for this._textContent you can just use `this._textContent.styles`.